ee/codegen/src/generators/expression.ts

```diff
@@ -101,6 +113,17 @@ export class Expression extends AstNode {
     );
   }
 
+  private isConstantValuePointer(lhs: AstNode): boolean {
+    return (
+      lhs instanceof NodeInput &&
+      !isNilOrEmpty(
+        lhs.nodeInputValuePointer.nodeInputValuePointerData.rules
+      ) &&
+      lhs.nodeInputValuePointer.nodeInputValuePointerData.rules[0]?.type ===
+        "CONSTANT_VALUE"
+    );
```

Comment:
[oos] this `rules[0]` is a bit spooky. I would've expected that we solved this at a per rule level instead of at the full value. I can't think of a counterexample though atm, so let's just keep a close 👀